### PR TITLE
2.0 version bump

### LINKED ISSUE #164: Ensure plugins work with Dashboards Node v14.18.2
Dashboards has upgraded Node to v14.18.2 and this is for plugin teams to verify that their plugins work with the new version upgrade. The Dashboards branch running the upgraded node is [feature/node14](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/feature/node14).

A list of things to check is:
* Your plugin's node version is bumped to 14.18.2
* Your plugin's `@types/node` package is bumped to `^14.17.32` if it's used
* Plugin builds and runs on Dashboards
* All of your plugin's tests pass
* Your GitHub workflows are not broken by the node version change and updated if they do
* Any sanity/smoke testing is successful

### LINKED ISSUE #221: [Hookup osd 2.0] AD Dashboards see 400 bad request issue
**Describe the bug**
Hook up AD Dashboards with OSD 2.0 see the following error:
```
{"statusCode":400,"error":"Bad Request","message":"[request body]: expected value of type [any] but got [undefined]"}
```

Screen Shot 2022-03-30 at 10 07 56 PM

Screen Shot 2022-03-30 at 10 08 36 PM

**To Reproduce**
Run OpenSearch 2.0 with AD engine 2.0
Clone AD Dashboards current main (v1.3) to OpenSearch Dashboards plugins folder
Modify packages.json (delete the packages that have the compatibility issue):
```
"@testing-library/jest-dom": "^5.11.4",
"@testing-library/react": "^11.0.4"
```
Change `opensearchDashboardsVersion` to 2.0.0 in opensearch_dashboards.json
Run `yarn osd bootstrap` and `yarn start` in OpenSearch Dashboards

**Expected behavior**
No issue

**Plugins**
AD

### LINKED ISSUE #222: [Hookup osd 2.0] AnnotationDomainTypes does not exists in @elastic/charts
**Describe the bug**
See any errors for all `AnnotationDomainTypes` imports. For example, `/anomaly-detection-dashboards-plugin/public/pages/AnomalyCharts/containers/AnomalyDetailsChart.tsx`

```
import {
  AnnotationDomainTypes,
```
will show the following error 
```
'"@elastic/charts"' has no exported member named 'AnnotationDomainTypes'. Did you mean 'AnnotationDomainType'?ts(2724)
specs.d.ts(691, 22): 'AnnotationDomainType' is declared here.
```

Then after hookup with OSD 2.0 and solved [400 Bad Request issue](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/issues/221) click `create detector` --> create a detector --> after detector initializing stage is done, click it. NO concept is shown.

Screen Shot 2022-03-30 at 11 13 09 PM

**To Reproduce**
Run OpenSearch 2.0 with AD engine 2.0
Clone AD Dashboards current main (v1.3) to OpenSearch Dashboards plugins folder
Modify packages.json (delete the packages that have the compatibility issue):

"@testing-library/jest-dom": "^5.11.4",
"@testing-library/react": "^11.0.4"
Change opensearchDashboardsVersion to 2.0.0 in opensearch_dashboards.json
Run yarn osd bootstrap and yarn start in OpenSearch Dashboards
After solve the [400 Bad Request issue](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/issues/221), click `create detector` --> create a detector --> after detector initializing stage is done, click it. NO concept is shown.

**Expected behavior**
No issue

**Plugins**
AD

### PR DESCRIPTION
**Pull request by amitgalitz**
### Description
1. Version bumped 1.3 to 2.0
2. added test environment that is now required for jest
3. removed jest-dom and react dependency in AD as we are now using a more up to date version from OSD
4. change wait() call to waitFor()
5. `AnnotationDomainTypes` changed to `AnnotationDomainType`
6. typescript upgrade made typing more restrictive so had to delete a line in get request which would make the body for get request `undefined` rather than `null`
7. Changed `SampleDataBox` component to use `EuiCard` instead of `ContentPanel`. This is because `betaBadgeLabel` was removed from `EuiPanel` which is used in `ContentPanel` component. This required a few changes in padding and adding a horizontal line below title manually.
8. Changed `SampleDataBox` to display in a grid so cards don't come off screen when minimizing it.
9. updated integ and unit test workflow to 2.0 and main branch on OSD

### Issues Resolved
resolves #221
resolves #222
resolves #164

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/AnomalyCharts/components/AlertsFlyout/__tests__/__snapshots__/AlertsFlyout.test.tsx.snap
Comment (ylwu-amzn) [inline public/pages/AnomalyCharts/components/AlertsFlyout/__tests__/__snapshots__/AlertsFlyout.test.tsx.snap:5]: Seems we changed a lot for alerts flyout. Looks quite simple now. Can you confirm if the `AlertsFlyout` snap is correct or not?

Comment (amitgalitz) [reply public/pages/AnomalyCharts/components/AlertsFlyout/__tests__/__snapshots__/AlertsFlyout.test.tsx.snap:5]: Investigating this now, ConfirmModal snap also had a lot removed. Need to see why testing isn't opening flyout/modal.

Comment (amitgalitz) [reply public/pages/AnomalyCharts/components/AlertsFlyout/__tests__/__snapshots__/AlertsFlyout.test.tsx.snap:5]: Still no clear answer on this. I opened an issue: https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/issues/231 and will continue investigating with the help on others on dashboard team. For the failing integ-test, the correct UI is still there but for some reason the aria-label for "Daily max" isn't present in the element. I will make a change to fix it on functional-test-repo after this is merged

Comment (amitgalitz) [reply public/pages/AnomalyCharts/components/AlertsFlyout/__tests__/__snapshots__/AlertsFlyout.test.tsx.snap:5]: Similar thing happened in ISM 2.0 update with their `CreateState` snap https://github.com/opensearch-project/index-management-dashboards-plugin/commit/7352cdbc9372dab2062538d5f292884f4a10e5a9#diff-7fde899dd6c89f4ee9fda19ed4cd3eff64cb1e0ba4e6e915d1e6631b42201351R4

**Review by ylwu-amzn — APPROVED** [review]
Approved this PR to unblock 2.0 development. You can fix failed IT in separate PR.

**Review by ananzh — APPROVED** [review]
nice. thx Amit

**Pull request merged**
